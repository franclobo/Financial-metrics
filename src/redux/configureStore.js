import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import reducer from './cryptocurrency';

const RootReducer = combineReducers({
  cryptos: reducer,
});

const store = createStore(RootReducer, applyMiddleware(thunk, logger));

export default store;
