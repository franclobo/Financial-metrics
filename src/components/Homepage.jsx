import { useDispatch, useSelector } from 'react-redux';
import { useState } from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FetchCryptos } from '../redux/cryptocurrency';

const Crypto = (props) => {
  const {
    name, change, price,
  } = props;

  return (
    <li>
      <NavLink to={`/${name}`}>
        <div className="cryptoList">
          <h2>
            {name}
          </h2>
          <p>
            change:
            {change}
          </p>
          <p>
            price:
            {price}
          </p>
        </div>
      </NavLink>
    </li>
  );
};

export default function Homepage() {
  const cryptos = useSelector((state) => state.cryptos);
  const dispatch = useDispatch();
  const [filterCrypto, setFilter] = useState([]);

  if (cryptos.length === 0) {
    dispatch(FetchCryptos());
  }

  const filterHandler = (e) => {
    if (e.target.value === 'Most Price') {
      setFilter(cryptos.filter((cryp) => cryp.price >= 3));
    } else if (e.target.value === 'Most Change') {
      setFilter(cryptos.filter((cryp) => cryp.change >= 1));
    } else if (e.target.value === 'Least Price') {
      setFilter(cryptos.filter((cryp) => cryp.price < 3));
    } else if (e.target.value === 'Least Change') {
      setFilter(cryptos.filter((cryp) => cryp.change < 1));
    } else {
      setFilter(cryptos);
    }
  };

  return (
    <div className="container">
      <div className="selector">
        <select name="filter" id="filter" onInput={(e) => filterHandler(e)}>
          <option defaultValue="Most Price">Most Price</option>
          <option value="Most Change">Most Change</option>
          <option value="Least Price">Least Price</option>
          <option value="Least Change">Least Change</option>
        </select>
      </div>
      <ul className="cryptos">
        {filterCrypto.length > 0
          && filterCrypto.map((crypto) => (
            <Crypto
              key={crypto.name}
              name={crypto.name}
              price={crypto.price}
              change={crypto.change}
            />
          ))}
        {filterCrypto.length === 0
          && cryptos.map((crypto) => (
            <Crypto
              key={crypto.name}
              name={crypto.name}
              price={crypto.price}
              change={crypto.change}
            />
          ))}
      </ul>
    </div>
  );
}

Crypto.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  change: PropTypes.number.isRequired,
};
