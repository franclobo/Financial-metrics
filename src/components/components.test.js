import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Homepage from './Homepage';

describe('Components Test', () => {
  test('Homepage Test', () => {
    const store = configureStore;
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Homepage />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
