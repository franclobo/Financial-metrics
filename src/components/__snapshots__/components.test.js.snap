// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Test Homepage Test 1`] = `
<div
  className="container"
>
  <div
    className="selector"
  >
    <select
      id="filter"
      name="filter"
      onInput={[Function]}
    >
      <option
        defaultValue="Most Price"
      >
        Most Price
      </option>
      <option
        value="Most Change"
      >
        Most Change
      </option>
      <option
        value="Least Price"
      >
        Least Price
      </option>
      <option
        value="Least Change"
      >
        Least Change
      </option>
    </select>
  </div>
  <ul
    className="cryptos"
  />
</div>
`;
